### ISSUE TITLE
Bug in add method of  "Perform a Union on Two Sets " challenge seed

### ISSUE THREAD

**Issue by nibble0101** [freeCodeCamp/freeCodeCamp#40444]
**Describe your problem and how to reproduce it:**

The [perform a union on two sets challenge](https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/perform-a-union-on-two-sets) in coding interview prep, data structures section requires one to create a method called `union` on the defined `Set` data structure in the challenge seed. This method should take another Set as an argument and return the union of the two sets, excluding any duplicate values. Essentially none of the solutions I submitted passed the tests. On closer inspection I noticed that the `set` objects used for testing the method are not sets. If you want to see what I mean, go to the challenge and add the following `union` method and click "run the tests".
```js
union(set){
    console.log(set);
    console.log(this.dictionary);
  }
```
You will see the following on the console.
```js
{ dictionary: { c: true, d: true }, length: 2 } // passed in to  union
{ a: true, b: true, c: true } // this.dictionary
```
None of the above are sets otherwise we wouldn't be having `true` appearing more than once in each of the above "sets".
I think this is caused by a bug in `add` method of [the project seed](https://github.com/freeCodeCamp/freeCodeCamp/blob/master/curriculum/challenges/english/10-coding-interview-prep/data-structures/perform-a-union-on-two-sets.md#--seed--). Essentially the `add` method in the project seed is:
```js
 add(element) {
    if (!this.has(element)) {
      this.dictionary[element] = true; // This should have been element
      this.length++;
      return true;
    }

    return false;
  }
```
Instead of adding the passed element, `true` is instead being added to the set. 

There also seems to be a contradiction between the instructions and what is actually expected in the test.
### The test instruction says

> The union of ["a", "b", "c"] and ["c", "d"] should return ["a", "b", "c", "d"].

From the example given, it appears an array is being passed to and returned from the union method  yet in actual sense an instance of `Set` is passed to and returned from the `union` method.

This issue of returning `keys` in the `values` method as highlighted in issue #40443 seems to have been carried to this challenge.
```js
values() {
    return Object.keys(this.dictionary);
  }
```

**Add a Link to the page with the problem:**

https://www.freecodecamp.org/learn/coding-interview-prep/data-structures/perform-a-union-on-two-sets

**Tell us about your browser and operating system:**
* Browser Name: N/A
* Browser Version: N/A
* Operating System: N/A

**If possible, add a screenshot here (you can drag and drop, png, jpg, gif, etc. in this box):**

**Comment by nhcarrigan** [comment]
Hey there @nibble0101! Would you be able to provide an example of code you wrote for the `union` method that was erroneously failing the tests? This way we can test against the changes made in the linked PR.

**Comment by nibble0101** [comment]
Hi @nhcarrigan . I stumbled across this issue because the solution I was submitting wasn't correct either. I misinterpreted  the statement:
> `The union of ["a", "b", "c"] and ["c", "d"] should return ["a", "b", "c", "d"].` 

to mean an array is passed to the `method` and then returning an array. However what I have just noticed is that as long as you are submitting a correct solution, the test will pass even if you leave the `add` and `values` methods bugs unfixed. Essentially when you submit a correct solution with the bugs unfixed, the created  `this.dictionary` object will look like
`this.dictionary = {'a': true, 'b':true', 'c': true, 'd': true }`. The `values` method is returning `Object.keys(this.dictionary)` which evaluates to `['a', 'b', 'c', 'd']`.  As a result when you look at the test assertions below, we are checking whether the passed elements have been set as keys of the `this.dictionary` object irrespective of whether the values are incorrectly or correctly set. So even if we had left the `this.dictionary` object to  be `this.dictionary = {'a': true, 'b':true', 'c': true, 'd': true }` as set by `add` method and the `values` method returning `Object.keys(this.dictionary)` the test would still pass because the key-value pairs are equal and the values have been ignored in the test. To me, it appears the linked PR fixes the issue. Though we can have the input of others as well.
 
```js
assert(
  (function () {
    var setA = new Set();
    var setB = new Set();
    setA.add('a');
    setA.add('b');
    setA.add('c');
    setB.add('c');
    setB.add('d');
    var unionSetAB = setA.union(setB);
    var final = unionSetAB.values();
    return (
      final.indexOf('a') !== -1 &&
      final.indexOf('b') !== -1 &&
      final.indexOf('c') !== -1 &&
      final.indexOf('d') !== -1 &&
      final.length === 4
    );
  })()
);
```
